src/net/metaplaylist/isobmff_patcher.ts

```diff
@@ -0,0 +1,461 @@
+      // Data offset present
+      if (flags & 0x1) {
+        dataOffsetPresent = true;
+        pos += 4;
+      }
+      // First sample flags present
+      if (flags & 0x4) {
+          pos += 4;
+      }
+      const sampleDurationPresent = flags & 0x100;
+      const sampleSizePresent = flags & 0x200;
+      const sampleFlagsPresent = flags & 0x400;
+      const sampleCompTimePresent = flags & 0x800;
+      let duration = 0;
+      for (let i = 0; i<sampleCount; i++) {
+        if (sampleDurationPresent) {
+            duration += be4toi(input, pos);
+            pos += 4;
+        }
+        if (sampleSizePresent) {
+          pos += 4;
+        }
+        if (sampleFlagsPresent) {
+            pos += 4;
+      }
+        if (sampleCompTimePresent) {
+            pos += 4;
+        }
+      }
+      this.duration = duration;
+      // Modify data_offset
+      let output = input.subarray(0, 16);
+      if (dataOffsetPresent && this.sizeChange > 0) {
+        let offset = be4toi(input, 16);
+        offset += this.sizeChange;
+        output = concat(output, itobe4(offset)); // func
+      }
+      else {
+        output = concat(output, input.subarray(16,20));
+      }
+      output = concat(output, input.subarray(20, input.length));
+      return output;
+    }
+
+    /**
+     * Process sidx data and add to output.
+     * @param {Uint8Array} input
+     * @param {boolean} keepSidx
+     */
+    sidx(input: Uint8Array, patchSidx: boolean): Uint8Array {
+      let output = new Uint8Array(0);
+      if (!patchSidx) {
+        return input;
+      }
+      let earliestPresentationTime;
+      let firstOffset;
+      const version = input[8];
+      const timescale = be4toi(input, 16);
+      if (version === 0) {
+        // Changing sidx version to 1
+        const size = be4toi(input, 0);
+        const sidxSizeExpansion = 8;
+        output = concat(
+          itobe4(size + sidxSizeExpansion),
+          input.subarray(4,8),
+          new Uint8Array([1]),
+          input.subarray(9, 20));
+        earliestPresentationTime = be4toi(input, 20);
+        firstOffset = be4toi(input, 24);
+      }
+      else {
+        output = input.subarray(0, 20);
+        earliestPresentationTime = be8toi(input, 20);
+        firstOffset = be8toi(input, 28);
+      }
+      const newPresentationTime = earliestPresentationTime + timescale * this.offset;
+      output = concat(output, itobe8(newPresentationTime), itobe8(firstOffset));
+      const suffixOffset = version === 0 ? 28 : 36;
+      output = concat(output, input.subarray(suffixOffset, input.length));
+      return output;
+    }
+
+    /**
+     * Generate new timestamps for tfdt and change size of boxes above if needed.
+     * Try to keep in 32 bits if possible.
+     * @param input
+     */
+    tfdt(input: Uint8Array): Uint8Array{
+      const version = input[8];
+      const tfdtOffset = this.offset;
+      let newBaseMediaDecodeTime;
+      let output = new Uint8Array(0);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this useless assignment to local variable "output". ([typescript:S1854](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1854))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=178&issues=AWNyk0oJTmDVX2bfxyay&open=AWNyk0oJTmDVX2bfxyay)

[ISSUE_KEY]: AWNyk0oJTmDVX2bfxyay